### PR TITLE
Handling time out message error, we have to think about handling the …

### PR DESCRIPTION
**Pull request by daividvleal**
…messeages in a better way.

### CODE REVIEW

File: app/src/main/kotlin/com/br/myfavoritehero/base/BaseViewModel.kt [hunk 1]
```diff
@@ -41,11 +42,20 @@ abstract class BaseViewModel : ViewModel(), LifecycleObserver {
                     HttpURLConnection.HTTP_BAD_METHOD -> {
                         errorResponse.extra = R.string.unknow_error
```
Comment (MatheusUehara) [inline app/src/main/kotlin/com/br/myfavoritehero/base/BaseViewModel.kt]: This conditional is needed? else condition handle this....

Comment (MatheusUehara) [reply app/src/main/kotlin/com/br/myfavoritehero/base/BaseViewModel.kt]: HttpURLConnection.HTTP_CONFLICT
                    HttpURLConnection.HTTP_NOT_FOUND
                    HttpURLConnection.HTTP_BAD_METHOD

File: app/src/main/kotlin/com/br/myfavoritehero/util/Constants.kt [hunk 2]
```diff
@@ -35,5 +35,7 @@ object Constants {
 
     const val ERROR_NOT_CONNECTED = 321
     const val MESSAGE_ERROR_NOT_CONNECTED = "Parece que você não tem uma conexão estável com a internet."
+    const val MESSAGE_ERROR_TIME_OUT = "Não estamos conseguindo nos conectar com a Marvel, acho que os vingadores estão com problemas! Só voltaremos mais tarde! O que você pode fazer para salvar o mundo enquanto isso?"
```
Comment (MatheusUehara) [inline app/src/main/kotlin/com/br/myfavoritehero/util/Constants.kt:40]: This is better in Strings file in case of future translations.. :D

**Review by MatheusUehara — CHANGES_REQUESTED** [review]

File: app/src/main/kotlin/com/br/myfavoritehero/util/Constants.kt (hunk 2 above)
Comment (daividvleal) [reply app/src/main/kotlin/com/br/myfavoritehero/util/Constants.kt:40]: it's already there, but this one is needed to not generate a linkage to the application context on BaseViewModel. Should be a better way to solve this, but for now its ok.

File: app/src/main/kotlin/com/br/myfavoritehero/base/BaseViewModel.kt (hunk 1 above)
Comment (daividvleal) [reply app/src/main/kotlin/com/br/myfavoritehero/base/BaseViewModel.kt]: Yes, the server can change the type request and the app should handle that.

Comment (MatheusUehara) [reply app/src/main/kotlin/com/br/myfavoritehero/base/BaseViewModel.kt]: I mean the else condition makes the same action of this ones:
                HttpURLConnection.HTTP_CONFLICT
                HttpURLConnection.HTTP_NOT_FOUND
                HttpURLConnection.HTTP_BAD_METHOD
if we don't do different action for each case this conditions don't make sense :)

**Comment by MatheusUehara** [comment]
Solve conflicts

**Review by MatheusUehara — APPROVED** [review]

**Pull request merged**